pkg/clusteragent/autoscaling/workload/controller_test.go

```diff
@@ -228,3 +235,51 @@ func TestLeaderCreateDeleteRemote(t *testing.T) {
 	f.RunControllerSync(true, "default/dpa-0")
 	assert.Len(t, f.store.GetAll(), 0)
 }
+
+func TestLeaderCreateDeleteLocalHeap(t *testing.T) {
+	testTime := time.Now()
+	f := newFixture(t, testTime)
+
+	dpaSpec := datadoghq.DatadogPodAutoscalerSpec{
+		TargetRef: autoscalingv2.CrossVersionObjectReference{
+			Kind:       "Deployment",
+			Name:       "app-0",
+			APIVersion: "apps/v1",
+		},
+		// Local owner means .Spec source of truth is K8S
+		Owner: datadoghq.DatadogPodAutoscalerLocalOwner,
+	}
+
+	// Read newly created DPA
+	dpa, dpaTyped := newFakePodAutoscaler("default", "dpa-0", 1, time.Now().Add(-1*time.Hour), dpaSpec, datadoghq.DatadogPodAutoscalerStatus{})
+	dpa1, dpaTyped1 := newFakePodAutoscaler("default", "dpa-1", 1, time.Now(), dpaSpec, datadoghq.DatadogPodAutoscalerStatus{})
+	dpa2, dpaTyped2 := newFakePodAutoscaler("default", "dpa-2", 1, time.Now().Add(1*time.Hour), dpaSpec, datadoghq.DatadogPodAutoscalerStatus{})
+
+	f.InformerObjects = append(f.InformerObjects, dpa)
+	f.Objects = append(f.Objects, dpaTyped)
+
+	f.InformerObjects = append(f.InformerObjects, dpa1)
+	f.Objects = append(f.Objects, dpaTyped1)
+	f.RunControllerSync(true, "default/dpa-1")
+	// Check that DatadogPodAutoscaler object is inserted into heap
+	assert.Equal(t, 1, f.autoscalingHeap.MaxHeap.Len())
+	assert.Equal(t, "default/dpa-1", f.autoscalingHeap.MaxHeap.Peek().Key)
+	assert.Truef(t, f.autoscalingHeap.Keys["default/dpa-1"], "Expected dpa-1 to be in heap")
+
+	f.InformerObjects = append(f.InformerObjects, dpa2)
+	f.Objects = append(f.Objects, dpaTyped2)
+	// Check that multiple objects can be inserted with ordering preserved
+	f.RunControllerSync(true, "default/dpa-2")
+	assert.Equal(t, 2, f.autoscalingHeap.MaxHeap.Len())
+	assert.Equal(t, "default/dpa-2", f.autoscalingHeap.MaxHeap.Peek().Key)
+	assert.Truef(t, f.autoscalingHeap.Keys["default/dpa-1"], "Expected dpa-1 to be in heap")
+	assert.Truef(t, f.autoscalingHeap.Keys["default/dpa-2"], "Expected dpa-2 to be in heap")
+
+	f.RunControllerSync(true, "default/dpa-0")
```

Comment:
The test is named `CreateDelete` but I don't see a test on the delete part? For instance removing `dpa-1` should make `dpa-2` eligible.